### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by keneogu**
This project is a clone of the Enumerable methods of the Ruby programming language

```ruby
# The Enumerable methods
#my_each
#my_each_with_index
#my_select 
#my_all? 
#my_any?
#my_none?
#my_count
#my_map
#my_inject
```

### CODE REVIEW

**Review by Georjane — CHANGES_REQUESTED** [review]
## Status: Changes Required 🛑 🔧 
### Hi Team ✋🏽 
Very nice work you have done with the enumerable methods. Just the following few tests are failing. Most of them have are caused when the argument is a range. Kindly go through them:

- [x] enumerables `#my_each` when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- [x] enumerables `#my_each_with_inde`x when a block is given when self is a range returns the range itself after calling the given block once for each element in self
- [x] enumerables `#my_any?` when no block or argument is given returns false if at least one of the collection is not true
- [x] enumerables `#my_count` returns the number of items in enum through enumeration::range
- [x] enumerables `#my_count` counts the number of elements yielding a true value if a block is given

Feel free to reach out to me on slack if you have any questions, **_Witah Georjane_** 🤝 

Good luck on your project and Happy Coding 💻

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    elsif block_given?
+      my_each { |element| return false unless yield(element) }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  # my_any Enumberable method
+  def my_any?(argue = nil)
```
Comment (mcrd25) [inline enumerables.rb]: Your `my_any` should return false if at least one of the collection is not true when no block or argument is given
Currently, your method does not handle this scenario accurately

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    if argu
+      my_each { |val| return false if argu === val } # rubocop:disable Style/CaseEquality
+    else
+      my_each { |val| return false if val == true }
+    end
+    result
+  end
+
+  # my_count Enumberable method
+  def my_count(para = nil)
```
Comment (mcrd25) [inline enumerables.rb:80]: `my_range` should also return the number of items when a range is passed.
It does not currently handle this well

Comment (mcrd25) [reply enumerables.rb:80]: Screenshot from 2021-01-08 14-19-10

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required :mag: 

Hi Team :wave: 

Good job! Only a couple of changes and you should be fine.


### Required Changes
- Your `my_any` should return false if at least one of the collection is not true when no block or argument is given (Correct this issue)
- your `my_count` does not handle ranges well.

***Please makes changes and submit another review request.
If you have any questions, let me know here or email (preferably) or on slack, and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

If you liked my review, consider upskilling me on LinkedIn. And please do connect. [My LinkedIn](https://www.linkedin.com/in/mayadouglas/)

*Happy coding 😄,*
**[--Maya Douglas](mayacdouglas.com)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by ghost — APPROVED** [review]
# Status: Approved 🎉 

**Congratulations team and good luck in your next project 😃.**

**Pull request merged**